app/Http/Controllers/SiteController.php

```diff
@@ -0,0 +1,258 @@
+ */
+class SiteController extends Controller
+{
+    /**
+     * SiteController constructor.
+     *
+     * @param APIService        $api
+     * @param SubscriberService $subscriber
+     * @param ContractService   $contract
+     * @param Validation          $validation
+     */
+    public function __construct(
+        APIService $api,
+        SubscriberService $subscriber,
+        ContractService $contract,
+        Validation $validation
+    ) {
+        $this->api        = $api;
+        $this->subscriber = $subscriber;
+        $this->contract   = $contract;
+        $this->validation = $validation;
+    }
+
+    /**
+     * @param Request $request
+     *
+     * @return \Illuminate\Contracts\View\Factory|\Illuminate\View\View
+     */
+    public function home(Request $request)
+    {
+        $email     = $request->input('email');
+        $source    = $request->input('source');
+        $groups    = $this->api->corporate_group();
+        $countries = $this->api->getCountryList();
+
+        return view(
+            'index',
+            compact(
+                'countries',
+                'groups',
+                'email',
+                'source'
+            )
+        );
+    }
+
+    /**
+     * Saves subscriber
+     *
+     * @param Request             $request
+     * @param ConfirmationService $confirm
+     *
+     * @return \Illuminate\Contracts\View\Factory|\Illuminate\Http\RedirectResponse|\Illuminate\View\View
+     */
+    public function postSubscriber(Request $request, ConfirmationService $confirm)
+    {
+        $data = [];
+        $this->validate($request, $this->validation->rules());
+        $data['email']  = $request->input('email');
+        $data['token']  = generateToken($data['email']);
+        $data['source'] = $request->input('source');
+        $data['status'] = 0;
+        $data['group']  = [
+            'country'         => isAllSelected(
+                $request->input('all_country'),
+                $request->input
+                (
+                    'country'
+                )
+            ),
+            'corporate_group' => isAllSelected(
+                $request->input('all_corporate_group'),
+                $request->input('corporate_group')
+            ),
+        ];
+        try {
+            $subscriber = $this->subscriber->createSubscriber($data);
+            $confirm->sendConfirmationEmail($subscriber);
+            return view('thanks');
+        } catch (\Exception $e) {
+            return redirect()->route('home')->withInput()->with('message', 'Error has occurred, please try again !');
+        }
+    }
+
+    /**
+     * Confirms user subscription
+     *
+     * @param $email
+     * @param $token
+     *
+     * @return \Illuminate\Contracts\View\Factory|\Illuminate\View\View
+     */
+    public function confirm($email, $token)
+    {
+        if ($subscriber = isTokenValid($email, $token)) {
+            $subscriber->status = 1;
```

Comment:
Service class handles all the logic. You should write logics in controller.
Controller only get request data from view and pass to service class.